### DISCUSSION TITLE
How to use the nested where clause in Prisma?

### DISCUSSION THREAD

**Discussion by sidvenu** [prisma/prisma#10980] [Q&A]
I am trying to do the equivalent of left join + where on multiple tables on Prisma. Here's the code:
![image](https://user-images.githubusercontent.com/18738527/148346460-4918585a-90a9-4253-aca2-57830ccb67fe.png)

However, I am getting the error:
`Type '{ where: {}; }' is not assignable to type 'boolean | transactionArgs | undefined'.
  Object literal may only specify known properties, and 'where' does not exist in type 'transactionArgs'`

The documentation at https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting says this should be possible. What am I missing?

**Comment by austincrim** [comment]
Hey @sidvenu 👋,

What you are trying should definitely be possible! Would you mind sharing your `schema.prisma` so I can try to recreate your issue locally?

Thanks!

**Comment by sidvenu** [comment]
Here you go:
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model transaction {
  id                             Int                                     @id @default(autoincrement()) @db.UnsignedInt
  amount                         Int                                     @db.UnsignedInt
  userId                         Int                                     @db.UnsignedInt
  status                         String                                  @db.VarChar(16)
  createTimestamp                Int                                     @db.UnsignedInt
  user                           user                                    @relation(fields: [userId], references: [id], onUpdate: Restrict, map: "transaction_ibfk_1")
  @@index([userId], map: "userId")
}

model user {
  id                  Int                   @id @default(autoincrement()) @db.UnsignedInt
  phoneNumber         String                @unique(map: "phoneNumber") @db.VarChar(16)
  email               String?               @db.VarChar(128)
  rzpyCustId          String?               @db.VarChar(128)
  transaction         transaction[]
}

model user_repayment {
  id                Int                            @id @default(autoincrement()) @db.UnsignedInt
  completeTimestamp Int?                           @db.UnsignedInt
  userId            Int                            @db.UnsignedInt
  transactionId     Int                            @db.UnsignedInt
  amount            Int                            @db.UnsignedInt
  transaction       transaction                    @relation(fields: [transactionId], references: [id], onUpdate: Restrict, map: "user_repayment_ibfk_2")
  user              user                           @relation(fields: [userId], references: [id], onUpdate: Restrict, map: "user_repayment_ibfk_1")
  @@index([transactionId], map: "transactionId")
  @@index([userId], map: "userId")
}
```

**Reply by austincrim** [reply]
In your screenshot, you are selecting from a `user_repayment` model, could you include that as well?

**Reply by sidvenu** [reply]
I've included it now, please do check.

**Reply by sidvenu** [reply]
Hey @austincrim, sorry for the pester. Do check it out when you are free.

**Comment by samrathchadha** [comment]
I'm not sure if I fully understand your use case, but you're basically trying to fetch the transaction field for every  user_repayment model right?

If so, transaction is a relation which needs to be fetched via `include` and since there isn't where logic for `include` yet, which you could request for in the form of an [issue](https://github.com/prisma/prisma/issues). Since there isn't any where logic for include, i usually use `array.filter()` and select all the elements I want.

**Reply by samrathchadha** [reply]
My bad, this example might work for you: https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting

**Reply by sidvenu** [reply]
Thanks for the example, @0xsamrath. Unfortunately, the same error as in the screenshot posted above occurs. 'where' does not exist as a type in tables within select/include for me somehow.

**Comment by nmfrankel** [comment]
The docs show how the nested `where` clause should be inside the regular `where` clause.

Try something like this...
```js
const autodebits2 = await prisma.user_repayment.findMany({
  where: {
    transaction: {
      amount: {
        gt: 10
      }
    }
  },
  select: {
    transaction: {
      select: {
        id: true,
        amount: true
      }
    },
    user: {
      select: {
        phoneNumber: true,
        email: true,
        rzpyCustId: true
      }
    }
  }
})
```

**Reply by pantharshit00** [reply]
@sidvenu Were you able to solve this using above suggestion? Please let us know if this is still relevant.

**Reply by javierfuentesm** [reply]
@pantharshit00 for me it is not working any idea why ? 
![image](https://user-images.githubusercontent.com/19244527/204432703-51467e1a-047c-416b-bc78-3fe010f98e39.png)

**Reply by javierfuentesm** [reply]
im getting this error but is not really clear to me 
![image](https://user-images.githubusercontent.com/19244527/204433219-6a06b1eb-735b-4d08-972e-818926690f6d.png)

**Reply by pantharshit00** [reply]
Hello!

It would be great if you can open a [new discussion](https://github.com/prisma/prisma/discussions/new) along with a small reproduction so that we can try it out locally.

**Comment by Elee-Lawleit** [comment]
Same issue for me, 

```
prisma.meeting.findMany({
            where: {
               id: userId
            },
            include : {
                timeslot:  true
            }
   })
```
   
   this works but
   
  ```
 prisma.meeting.findMany({
            where: {
               id: userId
            },
            include : {
                timeslot:  {
                    where: {
                        date: new Date()
                    }
                }
            }
   })
```
   
   doesn't work. I'm pretty sure this is what it says in the docs
   
   date is a field inside timeslot table

**Reply by nurul3101** [reply]
@Elee-Lawleit Can you please create a new GitHub Discussion and share your schema file, so that we can reproduce and investigate?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
